src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,31 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
 
 public class AsyncRequestProcessor {
     private final Executor executor;
 
+    private final Map<String, UserData> cache = new ConcurrentHashMap<>();
+
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        try {
+            Thread.sleep(500);
```

Comment:
You need to sleep (simulate work/waiting) only if there is no value in the cache by specified key. If value is present then return immediately.